core/src/main/java/org/veriblock/sdk/blockchain/BitcoinBlockchain.java

```diff
@@ -371,25 +370,37 @@ private void checkDifficulty(BitcoinBlock block, StoredBitcoinBlock previous) th
                 return;
             }
 
-            BigInteger newTarget = BitcoinUtils.calculateNewTarget(
+            long newTarget = calculateNewTarget(
                     BitcoinUtils.decodeCompactBits(previous.getBlock().getBits()),
                     cycleStart.getBlock().getTimestamp(),
                     previous.getBlock().getTimestamp());
 
-            int accuracyBytes = (block.getBits() >>> 24) - 3;
-            long receivedTargetCompact = block.getBits();
-
-            // The calculated difficulty is to a higher precision than received, so reduce here.
-            BigInteger mask = BigInteger.valueOf(0xFFFFFFL).shiftLeft(accuracyBytes * 8);
-            newTarget = newTarget.and(mask);
-            long newTargetCompact = BitcoinUtils.encodeCompactBits(newTarget);
-
-            if (newTargetCompact != receivedTargetCompact) {
+            if (block.getBits() != newTarget) {
                 throw new VerificationException("Block does not match computed difficulty adjustment");
             }
         }
     }
 
+    private long calculateNewTarget(BigInteger current, int startTimestamp, int endTimestamp) {
```

Comment:
I think this method should either accept compactBits as an input and return compactBits as output or accept BigInteger target and return BigInteger target. We should not mix both.

Reply:
I agree